src/query-builder/SelectQueryBuilder.ts

```diff
@@ -2063,20 +2084,6 @@ export class SelectQueryBuilder<Entity extends ObjectLiteral>
                 })
             }
         } else {
-            let subQuery: string = ""
-            if (typeof entityOrProperty === "function") {
-                const subQueryBuilder: SelectQueryBuilder<any> = (
-                    entityOrProperty as any
-                )((this as any as SelectQueryBuilder<any>).subQuery())
-                this.setParameters(subQueryBuilder.getParameters())
-                subQuery = subQueryBuilder.getQuery()
-            } else {
-                subQuery = entityOrProperty
-            }
-            const isSubQuery =
-                typeof entityOrProperty === "function" ||
-                (entityOrProperty.substr(0, 1) === "(" &&
-                    entityOrProperty.substr(-1) === ")")
```

Comment:
Pulled this up above the `if/else` block since it's used on both conditions now.